src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IReturnStatement.cs

```diff
@@ -1613,5 +1613,321 @@ System.Collections.Generic.IEnumerable<int> F()
+        {
+            string source = @"
+class C
+{
+    private static int _value;
+    static int Method()
+    {
+        return LocalFunction();
+
+        static ref int LocalFunction()
+        {
+            /*<bind>*/return ref _value;/*</bind>*/
+        }
+    }
+}
+";
+            string expectedOperationTree = @"
+IReturnOperation (RefKind: Ref) (OperationKind.Return, Type: null) (Syntax: 'return ref _value;')
+  ReturnedValue: 
+    IFieldReferenceOperation: System.Int32 C._value (Static) (OperationKind.FieldReference, Type: System.Int32) (Syntax: '_value')
+      Instance Receiver: 
+        null
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+            VerifyOperationTreeAndDiagnosticsForTest<ReturnStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void RefReturn10()
+        {
+            string source = @"
+class C
+{
+    private int _value;
+    int Method()
+    {
+        return LocalFunction();
+
+        ref int LocalFunction()
+            /*<bind>*/=> ref _value/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IBlockOperation (1 statements) (OperationKind.Block, Type: null) (Syntax: '=> ref _value')
+  IReturnOperation (RefKind: Ref) (OperationKind.Return, Type: null, IsImplicit) (Syntax: '_value')
+    ReturnedValue: 
+      IFieldReferenceOperation: System.Int32 C._value (OperationKind.FieldReference, Type: System.Int32) (Syntax: '_value')
+        Instance Receiver: 
+          IInstanceReferenceOperation (ReferenceKind: ContainingTypeInstance) (OperationKind.InstanceReference, Type: C, IsImplicit) (Syntax: '_value')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+            VerifyOperationTreeAndDiagnosticsForTest<ArrowExpressionClauseSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void RefReturn11()
+        {
+            string source = @"
+class C
+{
+    private static int _value;
+    static int Method()
+    {
+        return LocalFunction();
+
+        static ref int LocalFunction()
+            /*<bind>*/=> ref _value/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IBlockOperation (1 statements) (OperationKind.Block, Type: null) (Syntax: '=> ref _value')
+  IReturnOperation (RefKind: Ref) (OperationKind.Return, Type: null, IsImplicit) (Syntax: '_value')
+    ReturnedValue: 
+      IFieldReferenceOperation: System.Int32 C._value (Static) (OperationKind.FieldReference, Type: System.Int32) (Syntax: '_value')
+        Instance Receiver: 
+          null
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+            VerifyOperationTreeAndDiagnosticsForTest<ArrowExpressionClauseSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
```

Comment:
>} [](start = 8, length = 1)

Please add tests with RefReadonly. Also please add tests for lambdas both statement and expression forms. Also please make sure to cover cases where there is a mismatch in ref-ness between signature and the return statement.

Reply:
> Please add tests with RefReadonly.

This is now added.

> Also please make sure to cover cases where there is a mismatch in ref-ness between signature and the return statement.

I believe this is now added for methods and local functions. I will add properties and events.

> Also please add tests for lambdas both statement and expression forms.

Can you review this list and let me know if it covers the expected forms?

* `() => [result]`
* `x => [result]`
* `(x, y) => [result]`
* `(int x, int y) => [result]`
* `() => { return [result]; }`
* `x => { return [result]; }`
* `(x, y) => { return [result]; }`
* `(int x, int y) => { return [result]; }`
* `delegate { return [result]; }`
* `delegate (int x) { return [result]; }`

Reply:
>> Also please add tests for lambdas both statement and expression forms.

>Can you review this list and let me know if it covers the expected forms?

The list looks good. I think it would be sufficient to cover scenarios covered for local functions, including mismatch in ref-ness. 

---
In reply to: [295324507](https://github.com/dotnet/roslyn/pull/36271#discussion_r295324507) [](ancestors = 295324507)